components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/api/builder/WSO2UserOperationEventPayloadBuilder.java

```diff
@@ -234,4 +301,44 @@ private Group buildGroup(Map<String, Object> properties, AbstractUserStoreManage
 
         return group;
     }
+
+    public enum RecoveryCredentialType {
+        PASSWORD, USERNAME, NONE
+    }
+
+    private static RecoveryCredentialType getRecoveryCredentialType(String scenario) {
+
+        if (scenario == null) {
+            return RecoveryCredentialType.NONE;
+        }
+
+        // Handle constant string-based scenarios first
+        if (POST_CREDENTIAL_UPDATE_BY_ADMIN.equals(scenario) || POST_CREDENTIAL_UPDATE_BY_USER.equals(scenario)) {
+            return RecoveryCredentialType.PASSWORD;
+        }
+
+        try {
+            RecoveryScenarios recoveryScenario = RecoveryScenarios.valueOf(scenario);
+
+            switch (recoveryScenario) {
+                case NOTIFICATION_BASED_PW_RECOVERY:
+                case QUESTION_BASED_PWD_RECOVERY:
+                case ASK_PASSWORD:
+                case ADMIN_FORCED_PASSWORD_RESET_VIA_EMAIL_LINK:
+                case ADMIN_FORCED_PASSWORD_RESET_VIA_OTP:
+                case TENANT_ADMIN_ASK_PASSWORD:
+                case PASSWORD_EXPIRY:
+                    return RecoveryCredentialType.PASSWORD;
+
+                case USERNAME_RECOVERY:
+                    return RecoveryCredentialType.USERNAME;
+
+                default:
+                    return RecoveryCredentialType.NONE;
```

Comment:
why this is needed. IMO, you should be able to depend on identity flow context now. saying this again, see how the 'action' property in the payload gets populated in pre update password action

Reply:
Fixed